server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -405,9 +402,8 @@ protected void doStop() {
             });
         }
         try {
-            if (latch.await(shardsClosedTimeout.seconds(), TimeUnit.SECONDS) == false) {
-                logger.warn("Not all shards are closed yet, waited {}sec - stopping service", shardsClosedTimeout.seconds());
-            }
+            // infinite wait
+            latch.await();
         } catch (InterruptedException e) {
             // ignore
```

Comment:
We should reinstate the interrupt flag on the thread here

```suggestion
        } catch (InterruptedException e) {
            // continue with shutdown
            Thread.currentThread().interrupt();
```